src/CMakeLists.txt

```diff
@@ -21,12 +21,16 @@ set(CMAKE_ASM_COMPILER  ${PROJECT_SOURCE_DIR}/src/yasm-wrapper)
 set(CMAKE_ASM_FLAGS "-f elf64")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -rdynamic -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -fPIC")
 
+if(NOT CMAKE_BUILD_TYPE) 
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Default BUILD_TYPE is RelWithDebInfo, other options are: Debug Release MinSizeRel.") 
+endif(NOT CMAKE_BUILD_TYPE)
+
 include(CheckCCompilerFlag)
-CHECK_C_COMPILER_FLAG("-Wp,-D_FORTIFY_SOURCE=2" HAS_FORTIFY_SOURCE)
+CHECK_C_COMPILER_FLAG("-Wp,-D_FORTIFY_SOURCE=2 -Werr" HAS_FORTIFY_SOURCE)
 if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
-if(HAS_FORTIFY_SOURCE)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wp,-D_FORTIFY_SOURCE=2")
-endif()
+  if(HAS_FORTIFY_SOURCE)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wp,-D_FORTIFY_SOURCE=2")
+  endif()
```

Comment:
> don't need the check for Debug anymore

@cbodley we still need it? as `CHECK_C_COMPILER_FLAG()` does not respect `CMAKE_BUILD_TYPE`, i think. and fortify cflags does not work with debug build.